### PR TITLE
openimageio/all: Add new versions and many new dependencies.

### PR DESCRIPTION
**Pull request by sh0**
Specify library name and version:  **openimageio/2.2.7.0**, **openimageio/2.2.18.0**, **openimageio/2.3.7.2**

Adds many new dependency libraries which are now available in Conan. The `with_dicom` and `with_raw` flag names and default values were left as-is to preserve compatibility.

Each version requires their own patch due to changes in CMake scripts.

There are some diamond dependency conflicts. I currently handle them by using `requires` with `override=True`. As far as I know this is has not been done in any conan center packages. So how do I handle this situation?

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by sh0** [comment]
The CI failes due to bugs in FFmpeg conanfile. Closing until PR #7265 or some other fix is merged.

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

**Review by SSE4 — APPROVED** [review]
patches are extremely heave... probably, we need to provide all these numerous CMake variables in dependecies

**Comment by sh0** [comment]
Yes, it is quite unforturnate that the patches have to be so big. Part of the problem is that OIIO has its own `Find*.cmake` files which are somewhat non-standard. Perhaps some of this can be patched in upstream by renaming some files and variables.

**Comment by uilianries** [comment]
> There are some diamond dependency conflicts. I currently handle them by using requires with override=True. As far as I know this is has not been done in any conan center packages. So how do I handle this situation?

We usually solve other dependencies versions. Override is not a good option to be used. Can you explain the conflict?

**Comment by sh0** [comment]
@uilianries Currently I get with all flags (including `with_dicom`, `with_raw` and `with_openvdb`) turned on following conflicts:

```
'ffmpeg/4.2.1' requires 'openssl/1.1.1l' while 'dcmtk/3.6.6' requires 'openssl/1.0.2u'
'libraw/0.20.2' requires 'jasper/2.0.33' while 'opencv/4.5.2' requires 'jasper/2.0.32'
'c-blosc/1.20.1' requires 'zstd/1.4.5' while 'libtiff/4.3.0' requires 'zstd/1.5.0'
```
Looks like `lcms` diamond conflict has gone away already.

**Comment by uilianries** [comment]
@Sh0 please, share your entire log, including command line. I can't reproduce your error. If they are warning messages, then `override` is still not required.

**Comment by sh0** [comment]
@uilianries The overrides are currently under in the `conanfile.py` `requirements()` so unless you commented them out you won't have any conflicts. After removing them I get this:

```
$ conan create -b missing -b outdated -o openimageio:with_dicom=True -o openimageio:with_raw=True -o openimageio:with_openvdb=True . openimageio/2.3.7.2@

Exporting package recipe
openimageio/2.3.7.2 exports: File 'conandata.yml' found. Exporting it...
openimageio/2.3.7.2 exports: Copied 1 '.yml' file: conandata.yml
openimageio/2.3.7.2 exports_sources: Copied 1 '.txt' file: CMakeLists.txt
openimageio/2.3.7.2 exports_sources: Copied 4 '.patch' files: 2.2.7.0-boost-filesystem-is-regular.patch, 2.3.7.2-cmake-targets.patch, 2.2.7.0-cmake-targets.patch, 2.2.18.0-cmake-targets.patch
openimageio/2.3.7.2: A new conanfile.py version was exported
openimageio/2.3.7.2: Folder: /home/sh0/.conan/data/openimageio/2.3.7.2/_/_/export
openimageio/2.3.7.2: Using the exported files summary hash as the recipe revision: 000927b78f5186f17ab4adf03aa3ab32 
openimageio/2.3.7.2: Exported revision: 000927b78f5186f17ab4adf03aa3ab32
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
os.distro=gentoo
os_build=Linux
[options]
openimageio:with_dicom=True
openimageio:with_openvdb=True
openimageio:with_raw=True
[build_requires]
[env]

ERROR: Conflict in ffmpeg/4.2.1:
    'ffmpeg/4.2.1' requires 'openssl/1.1.1l' while 'dcmtk/3.6.6' requires 'openssl/1.0.2u'.
    To fix this conflict you need to override the package 'openssl' in your root package.

```

**Comment by uilianries** [comment]
@sh0 Probably you have some outdated cached. I don't see it, actually, it works without override. Please, remove it from the recipe.

My log:

<details>
<summary>Details</summary>

$ conan create -b missing -b outdated -o openimageio:with_dicom=True -o openimageio:with_raw=True -o openimageio:with_openvdb=True . openimageio/2.3.7.2@
Exporting package recipe
openimageio/2.3.7.2 exports: File 'conandata.yml' found. Exporting it...
openimageio/2.3.7.2 exports: Copied 1 '.yml' file: conandata.yml
openimageio/2.3.7.2 exports_sources: Copied 1 '.txt' file: CMakeLists.txt
openimageio/2.3.7.2 exports_sources: Copied 3 '.patch' files: 2.2.18.0-cmake-targets.patch, 2.2.7.0-cmake-targets.patch, 2.3.7.2-cmake-targets.patch
openimageio/2.3.7.2: The stored package has not changed
openimageio/2.3.7.2: Exported revision: 6101216adb68d9c8ea55a1f0fd3c5bd4
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=10
os=Linux
os_build=Linux
[options]
openimageio:with_dicom=True
openimageio:with_openvdb=True
openimageio:with_raw=True
[build_requires]
[env]

WARN: opencv/4.5.2: requirement jasper/2.0.32 overridden by openimageio/2.3.7.2 to jasper/2.0.33 
WARN: dcmtk/3.6.6: requirement openssl/1.0.2u overridden by openimageio/2.3.7.2 to openssl/1.1.1l 
WARN: pulseaudio/14.2: requirement libalsa/1.2.4 overridden by ffmpeg/4.2.1 to libalsa/1.2.5.1 
WARN: openvdb/8.0.1: requirement boost/1.75.0 overridden by openimageio/2.3.7.2 to boost/1.76.0 
WARN: openvdb/8.0.1: requirement openexr/2.5.5 overridden by openimageio/2.3.7.2 to openexr/2.5.7 
WARN: c-blosc/1.20.1: requirement zstd/1.4.5 overridden by openvdb/8.0.1 to zstd/1.5.0 
ade/0.1.1f: Package is up to date
brotli/1.0.9: Package is up to date
bzip2/1.0.8: Package is up to date
eigen/3.3.9: Package is up to date
fmt/8.0.1: Package is up to date
giflib/5.2.1: Package is up to date
gtk/system: Package is up to date
jbig/20160605: Package is up to date
lcms/2.12: Package is up to date
libalsa/1.2.5.1: Package is up to date
libbacktrace/cci.20210118: Package is up to date
libcap/2.50: Package is up to date
libde265/1.0.8: Package is up to date
libdeflate/1.8: Package is up to date
libfdk_aac/2.0.2: Package is up to date
libiconv/1.16: Package is up to date
libjpeg/9d: Package is up to date
libmp3lame/3.100: Package is up to date
libsquish/1.15: Package is up to date
libvpx/1.10.0: Package is up to date
libwebp/1.2.0: Package is up to date
libx264/20191217: Package is up to date
libx265/3.4: Package is up to date
lz4/1.9.2: Package is up to date
ogg/1.3.4: Package is up to date
openh264/2.1.1: Package is up to date
openjpeg/2.4.0: Package is up to date
openssl/1.1.1l: Package is up to date
opus/1.3.1: Package is up to date
pugixml/1.11: Package is up to date
quirc/1.1: Package is up to date
snappy/1.1.8: Package is up to date
tbb/2020.3: Package is up to date
tinyxml/2.6.2: Package is up to date
tsl-robin-map/0.6.3: Package is up to date
vaapi/system: Package is up to date
vdpau/system: Package is up to date
xorg/system: Package is up to date
xz_utils/5.2.5: Package is up to date
yaml-cpp/0.7.0: Package is up to date
zlib/1.2.11: Package is up to date
zstd/1.5.0: Package is up to date
boost/1.76.0: Package is up to date
c-blosc/1.20.1: Package is up to date
flac/1.3.3: Package is up to date
hdf5/1.12.0: Package is up to date
jasper/2.0.33: Package is up to date
libheif/1.12.0: Package is up to date
libpng/1.6.37: Package is up to date
libtiff/4.3.0: Package is up to date
libxml2/2.9.12: Package is up to date
opencolorio/1.1.1: Package is up to date
openexr/2.5.7: Package is up to date
protobuf/3.17.1: Package is up to date
ptex/2.4.0: Package is up to date
vorbis/1.3.7: Package is up to date
dcmtk/3.6.6: Package is up to date
freetype/2.10.4: Package is up to date
libraw/0.20.2: Package is up to date
libsndfile/1.0.31: Package is up to date
opencv/4.5.2: Package is up to date
openvdb/8.0.1: Package is up to date
pulseaudio/14.2: Package is up to date
ffmpeg/4.2.1: Package is up to date
openimageio/2.3.7.2: WARN: Package binary is corrupted, removing: 784a2ca9f153172797a87b72557e25c80269e139
openimageio/2.3.7.2 (test package): Installing package
Requirements
    ade/0.1.1f from 'conancenter' - Cache
    boost/1.76.0 from 'conancenter' - Cache
    brotli/1.0.9 from 'conancenter' - Cache
    bzip2/1.0.8 from 'conancenter' - Cache
    c-blosc/1.20.1 from 'conancenter' - Cache
    dcmtk/3.6.6 from 'conancenter' - Cache
    eigen/3.3.9 from 'conancenter' - Cache
    ffmpeg/4.2.1 from 'conancenter' - Cache
    flac/1.3.3 from 'conancenter' - Cache
    fmt/8.0.1 from 'conancenter' - Cache
    freetype/2.10.4 from 'conancenter' - Cache
    giflib/5.2.1 from 'conancenter' - Cache
    gtk/system from 'conancenter' - Cache
    hdf5/1.12.0 from 'conancenter' - Cache
    jasper/2.0.33 from 'conancenter' - Cache
    jbig/20160605 from 'conancenter' - Cache
    lcms/2.12 from 'conancenter' - Cache
    libalsa/1.2.5.1 from 'conancenter' - Cache
    libbacktrace/cci.20210118 from 'conancenter' - Cache
    libcap/2.50 from 'conancenter' - Cache
    libde265/1.0.8 from 'conancenter' - Cache
    libdeflate/1.8 from 'conancenter' - Cache
    libfdk_aac/2.0.2 from 'conancenter' - Cache
    libheif/1.12.0 from 'conancenter' - Cache
    libiconv/1.16 from 'conancenter' - Cache
    libjpeg/9d from 'conancenter' - Cache
    libmp3lame/3.100 from 'conancenter' - Cache
    libpng/1.6.37 from 'conancenter' - Cache
    libraw/0.20.2 from 'conancenter' - Cache
    libsndfile/1.0.31 from 'conancenter' - Cache
    libsquish/1.15 from 'conancenter' - Cache
    libtiff/4.3.0 from 'conancenter' - Cache
    libvpx/1.10.0 from 'conancenter' - Cache
    libwebp/1.2.0 from 'conancenter' - Cache
    libx264/20191217 from 'conancenter' - Cache
    libx265/3.4 from 'conancenter' - Cache
    libxml2/2.9.12 from 'conancenter' - Cache
    lz4/1.9.2 from 'conancenter' - Cache
    ogg/1.3.4 from 'conancenter' - Cache
    opencolorio/1.1.1 from 'conancenter' - Cache
    opencv/4.5.2 from 'conancenter' - Cache
    openexr/2.5.7 from 'conancenter' - Cache
    openh264/2.1.1 from 'conancenter' - Cache
    openimageio/2.3.7.2 from local cache - Cache
    openjpeg/2.4.0 from 'conancenter' - Cache
    openssl/1.1.1l from 'conancenter' - Cache
    openvdb/8.0.1 from 'conancenter' - Cache
    opus/1.3.1 from 'conancenter' - Cache
    protobuf/3.17.1 from 'conancenter' - Cache
    ptex/2.4.0 from 'conancenter' - Cache
    pugixml/1.11 from 'conancenter' - Cache
    pulseaudio/14.2 from 'conancenter' - Cache
    quirc/1.1 from 'conancenter' - Cache
    snappy/1.1.8 from 'conancenter' - Cache
    tbb/2020.3 from 'conancenter' - Cache
    tinyxml/2.6.2 from 'conancenter' - Cache
    tsl-robin-map/0.6.3 from 'conancenter' - Cache
    vaapi/system from 'conancenter' - Cache
    vdpau/system from 'conancenter' - Cache
    vorbis/1.3.7 from 'conancenter' - Cache
    xorg/system from 'conancenter' - Cache
    xz_utils/5.2.5 from 'conancenter' - Cache
    yaml-cpp/0.7.0 from 'conancenter' - Cache
    zlib/1.2.11 from 'conancenter' - Cache
    zstd/1.5.0 from 'conancenter' - Cache
Packages
    ade/0.1.1f:82ef5eac51c38971dea2fd342dd55ddf2ddfbbc3 - Cache
    boost/1.76.0:a762cbb0f4f91f9ac21c111bfd58400aa5d97a0b - Cache
    brotli/1.0.9:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    bzip2/1.0.8:91a8b22c2c5a149bc617cfc06cdd21bf23b12567 - Cache
    c-blosc/1.20.1:7082a21d2bffe51b337188af2ae7ebbc799a57fa - Cache
    dcmtk/3.6.6:dd44179262219439ed72b211a9ce5bc08a99f57b - Cache
    eigen/3.3.9:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    ffmpeg/4.2.1:7447de550b3dfb97958a02edda5817891be13dab - Cache
    flac/1.3.3:2c7e03b511b6c441dcef1045ca0020a42702472e - Cache
    fmt/8.0.1:1d886ef72d27d7ab9a1a64be277b3d28a290f3e4 - Cache
    freetype/2.10.4:bca7b8880d98719d556dd526ce612be20a815922 - Cache
    giflib/5.2.1:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    gtk/system:cc5e5ca8b619d2dffaaeae1e4924ae3ce085a8f1 - Cache
    hdf5/1.12.0:779a7f01d974aeab8fec852d32e96a6acc144d6f - Cache
    jasper/2.0.33:ce9cb5f00627ab693c3105ab1d4ed3e83b803c96 - Cache
    jbig/20160605:5dedf41b83996f2663697c4c548617b81e83ca54 - Cache
    lcms/2.12:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    libalsa/1.2.5.1:51e96d1346424f5ef647616231906873cd225eb3 - Cache
    libbacktrace/cci.20210118:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    libcap/2.50:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    libde265/1.0.8:debbc949d62e702ed363ae971f4da3700ba47b0d - Cache
    libdeflate/1.8:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    libfdk_aac/2.0.2:82ef5eac51c38971dea2fd342dd55ddf2ddfbbc3 - Cache
    libheif/1.12.0:6ceaf9a413efc7f836c99c780ac18c3e1b071255 - Cache
    libiconv/1.16:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    libjpeg/9d:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    libmp3lame/3.100:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    libpng/1.6.37:d5b3dc27faecfb4eb94086722000dd65bb9e6bff - Cache
    libraw/0.20.2:7a50f47ff8122c703ad9831728f36a2b3a261fa6 - Cache
    libsndfile/1.0.31:73e8d5f68c1a6912169268c061ca0c85bc58797e - Cache
    libsquish/1.15:82ef5eac51c38971dea2fd342dd55ddf2ddfbbc3 - Cache
    libtiff/4.3.0:45decf2f265a9a6a01b959bafc21551542ef03bc - Cache
    libvpx/1.10.0:b1d9f4db657294ab9602aa4bf9dd8888d4e05066 - Cache
    libwebp/1.2.0:abc00ad0249e9242a1b4cfef4dc265e67c782e32 - Cache
    libx264/20191217:88afa90131970fa8c89c3d86dda39233cf00f89e - Cache
    libx265/3.4:dac1ea31cbddea2f2da8eb01b1dd7b42e5bd2ceb - Cache
    libxml2/2.9.12:f9bf12b05a917e3b694bc35eb5037c71fbf6b7ca - Cache
    lz4/1.9.2:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    ogg/1.3.4:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    opencolorio/1.1.1:9d46612d8d84f58ba6e0d4f98ac53c30daf39b08 - Cache
    opencv/4.5.2:14e9f2f6b52c6e7ef5989c879fff6733696cd437 - Cache
    openexr/2.5.7:2ce08bf790c58b729dda567426e810ed5e35e513 - Cache
    openh264/2.1.1:82ef5eac51c38971dea2fd342dd55ddf2ddfbbc3 - Cache
    openimageio/2.3.7.2:784a2ca9f153172797a87b72557e25c80269e139 - Build
    openjpeg/2.4.0:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    openssl/1.1.1l:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    openvdb/8.0.1:09f74ccbabd65cc943d4bae395917716d2885dd2 - Cache
    opus/1.3.1:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    protobuf/3.17.1:743717ea6fefbda10f96b18fde0adbf093af5f63 - Cache
    ptex/2.4.0:2ce08bf790c58b729dda567426e810ed5e35e513 - Cache
    pugixml/1.11:82ef5eac51c38971dea2fd342dd55ddf2ddfbbc3 - Cache
    pulseaudio/14.2:c63e3687eb4370c7bcc632afbe163773ea0d1d6f - Cache
    quirc/1.1:f97b68fa945d442919986cc08aedf1dfba4cdbb7 - Cache
    snappy/1.1.8:82ef5eac51c38971dea2fd342dd55ddf2ddfbbc3 - Cache
    tbb/2020.3:7788bf09de359a829e85159c89ae9d40002b69bd - Cache
    tinyxml/2.6.2:82ef5eac51c38971dea2fd342dd55ddf2ddfbbc3 - Cache
    tsl-robin-map/0.6.3:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    vaapi/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    vdpau/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    vorbis/1.3.7:6e4bc3ab815736fa7c637f569f101e1193ddb5da - Cache
    xorg/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xz_utils/5.2.5:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    yaml-cpp/0.7.0:82ef5eac51c38971dea2fd342dd55ddf2ddfbbc3 - Cache
    zlib/1.2.11:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache
    zstd/1.5.0:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Cache

Installing (downloading, building) binaries...
ade/0.1.1f: Already installed!
brotli/1.0.9: Already installed!
bzip2/1.0.8: Already installed!
bzip2/1.0.8: Appending PATH environment variable: /home/conan/.conan/data/bzip2/1.0.8/_/_/package/91a8b22c2c5a149bc617cfc06cdd21bf23b12567/bin
eigen/3.3.9: Already installed!
fmt/8.0.1: Already installed!
giflib/5.2.1: Already installed!
giflib/5.2.1: Appending PATH environment variable: /home/conan/.conan/data/giflib/5.2.1/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/bin
gtk/system: Already installed!
jbig/20160605: Already installed!
jbig/20160605: Appending PATH environment variable: /home/conan/.conan/data/jbig/20160605/_/_/package/5dedf41b83996f2663697c4c548617b81e83ca54/bin
lcms/2.12: Already installed!
libalsa/1.2.5.1: Already installed!
libbacktrace/cci.20210118: Already installed!
libcap/2.50: Already installed!
libde265/1.0.8: Already installed!
libde265/1.0.8: Appending PATH environment variable: /home/conan/.conan/data/libde265/1.0.8/_/_/package/debbc949d62e702ed363ae971f4da3700ba47b0d/bin
libdeflate/1.8: Already installed!
libfdk_aac/2.0.2: Already installed!
libiconv/1.16: Already installed!
libiconv/1.16: Appending PATH environment var: /home/conan/.conan/data/libiconv/1.16/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/bin
libjpeg/9d: Already installed!
libmp3lame/3.100: Already installed!
libsquish/1.15: Already installed!
libvpx/1.10.0: Already installed!
libwebp/1.2.0: Already installed!
libx264/20191217: Already installed!
libx265/3.4: Already installed!
lz4/1.9.2: Already installed!
ogg/1.3.4: Already installed!
openh264/2.1.1: Already installed!
openjpeg/2.4.0: Already installed!
openssl/1.1.1l: Already installed!
opus/1.3.1: Already installed!
pugixml/1.11: Already installed!
quirc/1.1: Already installed!
snappy/1.1.8: Already installed!
tbb/2020.3: Already installed!
tinyxml/2.6.2: Already installed!
tsl-robin-map/0.6.3: Already installed!
vaapi/system: Already installed!
vdpau/system: Already installed!
xorg/system: Already installed!
xz_utils/5.2.5: Already installed!
yaml-cpp/0.7.0: Already installed!
zlib/1.2.11: Already installed!
zstd/1.5.0: Already installed!
boost/1.76.0: Already installed!
c-blosc/1.20.1: Already installed!
flac/1.3.3: Already installed!
flac/1.3.3: Appending PATH environment variable: /home/conan/.conan/data/flac/1.3.3/_/_/package/2c7e03b511b6c441dcef1045ca0020a42702472e/bin
hdf5/1.12.0: Already installed!
jasper/2.0.33: Already installed!
libheif/1.12.0: Already installed!
libpng/1.6.37: Already installed!
libtiff/4.3.0: Already installed!
libxml2/2.9.12: Already installed!
libxml2/2.9.12: Appending PATH environment variable: /home/conan/.conan/data/libxml2/2.9.12/_/_/package/f9bf12b05a917e3b694bc35eb5037c71fbf6b7ca/bin
opencolorio/1.1.1: Already installed!
opencolorio/1.1.1: Appending PATH env var with: /home/conan/.conan/data/opencolorio/1.1.1/_/_/package/9d46612d8d84f58ba6e0d4f98ac53c30daf39b08/bin
openexr/2.5.7: Already installed!
protobuf/3.17.1: Already installed!
protobuf/3.17.1: Appending PATH environment variable: /home/conan/.conan/data/protobuf/3.17.1/_/_/package/743717ea6fefbda10f96b18fde0adbf093af5f63/bin
ptex/2.4.0: Already installed!
vorbis/1.3.7: Already installed!
dcmtk/3.6.6: Already installed!
dcmtk/3.6.6: Settings DCMDICTPATH environment variable: /home/conan/.conan/data/dcmtk/3.6.6/_/_/package/dd44179262219439ed72b211a9ce5bc08a99f57b/bin/share/dcmtk/dicom.dic
freetype/2.10.4: Already installed!
libraw/0.20.2: Already installed!
libsndfile/1.0.31: Already installed!
libsndfile/1.0.31: Appending PATH environment variable: /home/conan/.conan/data/libsndfile/1.0.31/_/_/package/73e8d5f68c1a6912169268c061ca0c85bc58797e/bin
opencv/4.5.2: Already installed!
openvdb/8.0.1: Already installed!
pulseaudio/14.2: Already installed!
ffmpeg/4.2.1: Already installed!
openimageio/2.3.7.2: WARN: Build folder is dirty, removing it: /home/conan/.conan/data/openimageio/2.3.7.2/_/_/build/784a2ca9f153172797a87b72557e25c80269e139
openimageio/2.3.7.2: Copying sources to build folder
openimageio/2.3.7.2: Building your package in /home/conan/.conan/data/openimageio/2.3.7.2/_/_/build/784a2ca9f153172797a87b72557e25c80269e139
openimageio/2.3.7.2: Generator cmake_find_package created Findpugixml.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibsquish.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findtsl-robin-map.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findfmt.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindHDF5.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindOpenColorIO.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindOpenCV.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindDCMTK.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findffmpeg.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindGIF.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibheif.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibraw.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindOpenVDB.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findptex.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlcms.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindJasper.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindBoost.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindTIFF.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindOpenEXR.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindFreetype.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findyaml-cpp.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindTinyXML.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindEigen3.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findquirc.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findgtk.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindProtobuf.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findade.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindTBB.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindLibXml2.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindOpenJPEG.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findopenh264.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibx264.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibx265.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibvpx.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibmp3lame.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findfdk-aac.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findpulseaudio.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findvaapi.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findvdpau.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibde265.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findc-blosc.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findzstd.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindJPEG.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindOpenSSL.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindBZip2.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibbacktrace.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibdeflate.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindLibLZMA.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findjbig.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindWebP.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindPNG.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindBrotli.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindIconv.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindALSA.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findxorg.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindSndFile.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlibcap.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findlz4.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindSnappy.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindZLIB.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindVorbis.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindOpus.cmake
openimageio/2.3.7.2: Generator cmake_find_package created Findflac.cmake
openimageio/2.3.7.2: Generator cmake_find_package created FindOgg.cmake
openimageio/2.3.7.2: Generator cmake created conanbuildinfo.cmake
openimageio/2.3.7.2: Aggregating env generators
openimageio/2.3.7.2: Calling build()
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- Detecting C compiler ABI info


</details>

**Comment by sh0** [comment]
@uilianries Looks like there were some outdated packages in cache indeed. Things sure are moving fast in conan...

**Comment by uilianries** [comment]
> @uilianries Looks like there were some outdated packages in cache indeed. Things sure are moving fast in conan...

@sh0 Indeed there are cases where build_requires and requires want the same package, but different versions. For those cases, Conan will fail and neither override can solve, only dual profile can solve. I though you had that case, but fortunately was just an outdated package.

File: recipes/openimageio/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -6,4 +6,3 @@ conan_basic_setup()
```
Comment (uilianries) [inline recipes/openimageio/all/test_package/CMakeLists.txt:6]: Update this file to:

```
cmake_minimum_required(VERSION 3.1)
project(test_package CXX)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} OpenImageIO::OpenImageIO)
```

File: recipes/openimageio/all/test_package/conanfile.py [hunk 2]
```diff
@@ -6,11 +6,13 @@ class TestPackageConan(ConanFile):
     generators = "cmake"
```
Comment (uilianries) [inline recipes/openimageio/all/test_package/conanfile.py:6]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

File: recipes/openimageio/all/patches/2.2.7.0-cmake-targets.patch [hunk 3]
```diff
@@ -145,20 +523,151 @@ index 083b077c..a7ec5cda 100644
...
+--- a/src/zfile.imageio/CMakeLists.txt
++++ b/src/zfile.imageio/CMakeLists.txt
+@@ -3,5 +3,5 @@
+ # https://github.com/OpenImageIO/oiio/blob/master/LICENSE.md
+ 
+ add_oiio_plugin (zfile.cpp
+-                 INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS}
+-                 LINK_LIBRARIES ${ZLIB_LIBRARIES})
++                 # INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS}
++                 LINK_LIBRARIES ZLIB::ZLIB)
```
Comment (uilianries) [inline recipes/openimageio/all/patches/2.2.7.0-cmake-targets.patch:673]: Not much to do here. All patches are big, but they make all dependencies work correctly with Conan.  :+1:

File: recipes/openimageio/all/test_package/CMakeLists.txt (hunk 1 above)
Comment (sh0) [reply recipes/openimageio/all/test_package/CMakeLists.txt:8]: Updated as suggested 👍

File: recipes/openimageio/all/test_package/conanfile.py (hunk 2 above)
Comment (sh0) [reply recipes/openimageio/all/test_package/conanfile.py]: Also added.

File: recipes/openimageio/all/patches/2.2.7.0-cmake-targets.patch (hunk 3 above)
Comment (sh0) [reply recipes/openimageio/all/patches/2.2.7.0-cmake-targets.patch:673]: Upstream could make our life somewhat easier if they used more target-based linking with standard names...

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM!

**Review by SSE4 — APPROVED** [review]

**Pull request merged**